src/s1reader/s1_reader.py

```diff
@@ -437,16 +437,80 @@ def burst_from_zip(zip_path: str, orbit_path: str, swath_num: int, pol: str = 'v
         raise ValueError(f"polarization not in {pols}")
 
     id_str = f'iw{swath_num}-slc-{pol}'
+
+    if not os.path.exists(path):
+        raise FileNotFoundError(f'{path} not found')
+    elif os.path.isdir(path):
+        return _burst_from_safe_dir(path, id_str, orbit_path)
+    elif os.path.isfile(path):
+        return _burst_from_zip(path, id_str, orbit_path)
+    else:
+        raise ValueError(f'{path} is unsupported')
+
+def _burst_from_zip(zip_path: str, id_str: str, orbit_path: str):
+    '''Find bursts in a Sentinel 1 zip file.
+
+    Parameters:
+    -----------
+    path : str
+        Path to zip file.
+    id_str: str
+        Identifcation of desired burst. Format: iw[swath_num]-slc-[pol]
+    orbit_path : str
+        Path the orbit file.
+
+    Returns:
+    --------
+    bursts : list
+        List of Sentinel1BurstSlc objects found in annotation XML.
+    '''
     with zipfile.ZipFile(zip_path, 'r') as z_file:
+        z_file_list = z_file.namelist()
+
         # find annotation file
-        f_annotation = [f for f in z_file.namelist() if _is_annotation_xml(f, id_str)]
+        f_annotation = [f for f in z_file_list if _is_zip_annotation_xml(f, id_str)]
         if not f_annotation:
-            raise ValueError(f"polarization {pol} not in SAFE: {zip_path}")
+            raise ValueError(f"burst {id_str} not in SAFE: {zip_path}")
         f_annotation = f_annotation[0]
 
         # find tiff file
-        f_tiff = [f for f in z_file.namelist() if 'measurement' in f and id_str in f and 'tiff' in f][0]
-        f_tiff = f'/vsizip/{zip_path}/{f_tiff}'
+        f_tiff = [f for f in z_file_list
+                  if 'measurement' in f and id_str in f and 'tiff' in f]
+        f_tiff = f'/vsizip/{zip_path}/{f_tiff[0]}' if f_tiff else ''
 
         bursts = burst_from_xml(f_annotation, orbit_path, f_tiff, z_file.open)
         return bursts
+
+def _burst_from_safe_dir(safe_dir_path: str, id_str: str, orbit_path: str):
+    '''Find bursts in a Sentinel 1 SAFE structured directory.
+
+    Parameters:
+    -----------
+    path : str
+        Path to SAFE directory.
+    id_str: str
+        Identifcation of desired burst. Format: iw[swath_num]-slc-[pol]
+    orbit_path : str
+        Path the orbit file.
+
+    Returns:
+    --------
+    bursts : list
+        List of Sentinel1BurstSlc objects found in annotation XML.
+    '''
+
+    # find annotation file
+    annotation_list = os.listdir(f'{safe_dir_path}/annotation')
+    f_annotation = [f for f in annotation_list if id_str in f]
+    if not f_annotation:
+        raise ValueError(f"burst {id_str} not in SAFE: {safe_dir_path}")
+    f_annotation = f'{safe_dir_path}/annotation/{f_annotation[0]}'
+
+    # find tiff file
+    measurement_list = os.listdir(f'{safe_dir_path}/measurement')
```

Comment:
For the case that we may only download the annotations, we don't have measurement in the path. Therefore we would get error like this one:

```
Traceback (most recent call last):
  File "/u/aurora-r0/fattahi/burst_map/./test_read.py", line 14, in <module>
    bursts = s1reader.load_bursts(zip_path, orbit_path, swath_num, pol)
  File "/home/fattahi/tools/anaconda3/envs/burst_id/lib/python3.10/site-packages/s1reader/s1_reader.py", line 444, in load_bursts
    return _burst_from_safe_dir(path, id_str, orbit_path)
  File "/home/fattahi/tools/anaconda3/envs/burst_id/lib/python3.10/site-packages/s1reader/s1_reader.py", line 510, in _burst_from_safe_dir
    measurement_list = os.listdir(f'{safe_dir_path}/measurement')
FileNotFoundError: [Errno 2] No such file or directory: 'xmls/S1A_IW_SLC__1SDV_20190330T015159_20190330T015226_026561_02FA29_6720.SAFE/measurement'
```

Reply:
Perhaps just check if `f'{safe_dir_path}/measurement'` exists and then go through the next few lines. If the measurement does not exist, let's throw a warning that measurement does not exist and proceed because we still can read the metadata.